artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/JournalStorageManager.java

```diff
@@ -432,7 +432,7 @@ public void freeDirectBuffer(ByteBuffer buffer) {
       journalFF.releaseBuffer(buffer);
    }
 
-   public long storePendingLargeMessage(final long messageID, long recordID) throws Exception {
```

Comment:
I'm not sure this is a good idea, this is introducing a blocking operation.

Reply:
which will block any netty processing.

Reply:
although delete itself is already blocking.. so I think it's fine.

Reply:
There is diff between a memory block and an waiting on disk io.

Can this not be async with a callback with a response later?

Reply:
I am sending an alternate PR.

Reply:
@qihongxu I fixed it without a blocking operation through a different PR. Please let me know if you see any issues.

Thanks

@michaelandrepearce  ^^